kv/kv.go

```diff
@@ -235,6 +235,12 @@ type Transaction interface {
 	SetDiskFullOpt(level kvrpcpb.DiskFullOpt)
 	// clear allowed flag
 	ClearDiskFullOpt()
+
+	// GetMemDBCheckpoint gets the transaction's memDB checkpoint.
+	GetMemDBCheckpoint() *tikv.MemDBCheckpoint
```

Comment:
I don't think it's a good practice to introduce more `tikv` interface in `kv` package, all interfaces should be abstracted.

But since we have such cases already like:

https://github.com/pingcap/tidb/blob/7f023bd61b6228a40d626f49e1cb0c37c8222b85/kv/kv.go#L436

We don't have to solve it in this PR.